gr-blocks/lib/moving_average_impl.cc

```diff
@@ -87,21 +150,18 @@ int moving_average_impl<T>::work(int noutput_items,
     if (d_updated) {
         d_length = d_new_length;
         d_scale = d_new_scale;
+        std::fill(std::begin(d_scales), std::end(d_scales), d_scale);
         this->set_history(d_length);
         d_updated = false;
         return 0; // history requirements might have changed
     }
 
-    const T* in = (const T*)input_items[0];
-    T* out = (T*)output_items[0];
+    const T* in = static_cast<const T*>(input_items[0]);
+    T* out = static_cast<T*>(output_items[0]);
 
-    unsigned int num_iter =
-        (unsigned int)((noutput_items > d_max_iter) ? d_max_iter : noutput_items);
+    const unsigned int num_iter = std::min(noutput_items, d_max_iter);
     if (d_vlen == 1) {
-        T sum = in[0];
-        for (int i = 1; i < d_length - 1; i++) {
-            sum += in[i];
-        }
+        T sum = std::accumulate(&in[0], &in[d_length - 1], T{});
```

Comment:
`volk_32f_accumulator_s32f()`

[1] https://www.libvolk.org/doxygen/volk_32f_accumulator_s32f.html

Reply:
`std::accumulate` I think vectorizes pretty well already, according to my experimentation.

Sure, volk can do more at runtime to choose the best kernel. For now though I'd prefer to leave this for a future PR.

Reply:
`std::accumulate` does vectorize when applied to integer types, but not when using float types – unless you use `-ffast-math` (according to my experimentation).